src/tracker.rs

```diff
@@ -184,16 +223,25 @@ impl Tracker {
         }
     }
 
+    pub fn has_pointer(&self, point_offset: PointOffset) -> bool {
+        self.pending_updates.contains_key(&point_offset) || self.get(point_offset).is_some()
+    }
+
+    pub fn update(&mut self, point_offset: PointOffset, value_pointer: ValuePointer) {
+        self.pending_updates.insert(point_offset, value_pointer);
+    }
+
     /// Set value at the given point offset
     /// If the point offset is larger than the current length, the mapping is resized
-    pub fn set(&mut self, point_offset: PointOffset, value_pointer: ValuePointer) {
+    fn set(&mut self, point_offset: PointOffset, value_pointer: ValuePointer) {
```

Comment:
Having this functionality as a separate function might be confusing, as it is never intended to be used outside of flush. But currently it is used in tests, which doesn't match expected usage pattern

Reply:
In the end I decided to keep `set` and inline the persistance into `persist_pointer`